crypto/stark/src/rational_expression.rs

```diff
@@ -64,6 +66,24 @@ impl RationalExpression {
         };
         f(e)
     }
+
+    pub fn substitute_claim(&self, claim_polynomials: &[DensePolynomial]) -> RationalExpression {
+        let f = |x: RationalExpression| {
+            match x {
+                RationalExpression::ClaimPolynomial(i, a) => {
+                    RationalExpression::Polynomial(
```

Comment:
You can put `use RationalExpression::*` at the start of the function. It might make this a bit more readable.